BaseStation/BaseServer/BaseServer.cs

```diff
@@ -196,6 +199,24 @@ private static List<float> ConvertToFloatArray(Packet data)
             return ret;
         }
 
+        private static List<double> ConvertToDoubleArray(Packet data)
```

Comment:
All instances of use of ConvertToFloatArray should be removed and replaced with this? This is basically correct duplicated code and code that used ConvertToFloatArray should be able to cast down.

Reply:
this one divides it into arrays of length 8 which are needed for doubles. you would be losing half of your data if you tried to use the this version of the converting on a string of floats. it's a necessary duplication.

Reply:
it's possible to factor out some of it however

Reply:
You can factor out basically all of it and just make a second parameter with how many values to unpack!

Reply:
I can't factor out the utildata.tofloat and untildata.todouble